app/viewmodels/TradeJourneyStatusesViewModel.scala

```diff
@@ -106,34 +127,33 @@ object TradeJourneyStatusesViewModel {
       None
     }
 
-  private def getJourneyStatus(journey: Journey, conditionalCompletedJourney: Option[Journey])(implicit
-      journeyStatuses: TradesJourneyStatuses): JourneyStatus =
-    statusFromCompletedState(getCompletedState(journeyStatuses, journey)) match {
-      case NotStarted            => conditionalJourneyIsPassed(conditionalCompletedJourney)
-      case status: JourneyStatus => status
+  private def dependentJourneyIsInProgressOrCompleted(dependentJourney: Journey, checkOnlyCompleted: Boolean = false)(implicit
+      journeyStatuses: TradesJourneyStatuses): Boolean =
+    getJourneyStatus(dependentJourney) match {
+      case Completed | InProgress if !checkOnlyCompleted => true
+      case Completed                                     => true
+      case _                                             => false
```

Comment:
it's good idea to not use `_` if it's a small number of elements.

Then if we add new sealed value, the code will not compile until we consciously change each of these pattern matches. Otherwise, when we add new value the code will compile which is not good, because may lead to some bugs.